lib/spack/spack/bootstrap.py

```diff
@@ -20,17 +25,33 @@
 from spack.util.environment import EnvironmentModifications
 
 
+def spec_for_current_python():
+    """For bootstrapping purposes we are just interested in the Python
+    minor version (all patches are ABI compatible with the same minor)
+    and on whether ucs4 support has been enabled for Python 2.7
+
+    See:
+      https://www.python.org/dev/peps/pep-0513/
+      https://stackoverflow.com/a/35801395/771663
+    """
+    version_str = '.'.join(str(x) for x in sys.version_info[:2])
+    variant_str = ''
+    if sys.version_info[0] == 2 and sys.version_info[1] == 7:
+        unicode_size = sysconfig.get_config_var('Py_UNICODE_SIZE')
+        variant_str = '+ucs4' if unicode_size == 4 else '~ucs4'
+
+    spec_fmt = 'python@{0} {1}'
+    return spec_fmt.format(version_str, variant_str)
+
+
 @contextlib.contextmanager
 def spack_python_interpreter():
     """Override the current configuration to set the interpreter under
     which Spack is currently running as the only Python external spec
     available.
     """
-    python_cls = type(spack.spec.Spec('python').package)
     python_prefix = os.path.dirname(os.path.dirname(sys.executable))
-    externals = python_cls.determine_spec_details(
-        python_prefix, [os.path.basename(sys.executable)])
-    external_python = externals[0]
+    external_python = spec_for_current_python()
```

Comment:
We're  losing a lot of information here compared to `determine_spec_details`. How confident are you that we won't actually need that information?

Reply:
The information here permits to distinguish among the interpreters provided in the `manylinux1` image:
```console
[root@90e0a523cd26 /]# ls /opt/python/
cp27-cp27m  cp27-cp27mu  cp35-cp35m  cp36-cp36m  cp37-cp37m  cp38-cp38  cp39-cp39
```
which, to my understanding of [PEP-513](https://www.python.org/dev/peps/pep-0513/#the-manylinux1-policy) are the only one we need to care about for Python on `x86_64` and versions 2.7, 3.5-39. In principle I think Python 2.6 should be treated the same as Python 2.7 (and that is not accounted for in this PR) but I think that for that old Python version we may just stick to whatever build the CentOS 6 image pulls in via yum.